### PR TITLE
Reorganize samples to being onboarding to https://docs.microsoft.com/samples/.

### PR DESCRIPTION
**Pull request by cgranade**
This draft PR begins reorganizing samples to make the categorization more clear (e.g.: placing chemistry and numerics samples under the `samples/` folder), and to add metadata to "Getting Started" samples needed for inclusion in https://docs.microsoft.com/samples/.

**TODO**:
- [x] Make sure solution files for each folder under `samples/` exist and are correct.
- [x] Update README.md at the root, and in each folder under `samples/`.

### CODE REVIEW

File: samples/getting-started/intro-to-iqsharp/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+page_type: sample
+languages: [qsharp]
+products: [qdk]
+---
+
+# Intro to IQ# #
```
Comment (natke) [inline samples/getting-started/intro-to-iqsharp/README.md]: I would suggest renaming this to something like: "Intro to Q# Jupyter notebooks"

File: samples/getting-started/teleportation/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+﻿---
+page_type: sample
+languages: [qsharp, python, csharp]
+products: [qdk]
+---
+
+# Quantum Teleportation Sample #
```
Comment (natke) [inline samples/getting-started/teleportation/README.md]: We don't need the pounds around the headings, just at the beginning. Having said that, I'm not sure if they will be rendered as # or ignored by the markdig renderer.

File: samples/getting-started/intro-to-iqsharp/README.md (hunk 1 above)
Comment (cgranade) [reply samples/getting-started/intro-to-iqsharp/README.md]: Good point, will go on and rename, thanks!

File: samples/getting-started/teleportation/README.md (hunk 2 above)
Comment (cgranade) [reply samples/getting-started/teleportation/README.md]: I always learned the style with `#` at both ends, but I'm fine either way; happy to normalize to only leading `#` if that's more consistent.

File: samples/getting-started/measurement/Host.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+{
+    class Program
+    {
+
+        static void Main(string[] args)
+        {
+
+            using (var qsim = new QuantumSimulator())
+            {
+                RunQuantumMain.Run(qsim).Wait();
```
Comment (bettinaheim) [inline samples/getting-started/measurement/Host.cs]: I am really looking forward to getting rid of the need for this!

**Pull request merged**